### ISSUE TITLE
sentry-rails depends on ActiveRecord

### ISSUE THREAD

**Issue by thibaudgg** [getsentry/sentry-ruby#1345]
We were trying to install Sentry to a small and custom Rails application that doesn't make use of ActiveRecord and we were stuck not being able to send any Sentry event.

After some debugging, we find out that the `sentry/rails/background_worker` is expecting ActiveRecord presence and sending the event is failing silently. 

Removing the `ActiveRecord::Base.connection_pool.with_connection` block fixes the issue and events are sent successfully.

It would be nice to check for AR presence here.

``` ruby
# sentry/rails/background_worker.rb
module Sentry
  class BackgroundWorker
    alias_method :original_perform, :perform

    def perform(&block)
      @executor.post do
        if Object.const_defined?('ActiveRecord')
          # make sure the background worker returns AR connection if it accidentally acquire one during serialization
          ActiveRecord::Base.connection_pool.with_connection do
            block.call
          end
        else
          block.call
        end
      end
    end
  end
end
```

**To Reproduce**
Any Rails app not using ActiveRecord.

**Expected behavior**
Being able to send Sentry event.

**Actual behavior**
Send Sentry event is not working.

**Environment**
 - Ruby Version: 2.6.6
 - SDK Version: 4.3.0
 - Rails 6.0

**Comment by st0012** [comment]
@anderscarling thanks for reporting this, we've released `sentry-rails 4.3.1` to address the issue 🙂

**Comment by thibaudgg** [comment]
Thanks @st0012! 👍🏻
